### ISSUE TITLE
Could not pass event PlayerLoginEvent

### ISSUE THREAD

**Issue by andrewleppala** [EverNife/EverNifeCore#3]
Server Platform: `mohist-1.16.5-1136-server`
Spigot API: `1.16.5-R0.1-SNAPSHOT`
Forge Version: `36.2.39`

Mods:
-architectury-1.32.66
- betterfpsdist-1.2
- BiomesOPlenty-1.16.5-13.1.0.487
- byg-1.3.5
- does_it_tick-1.16.5_v2.0.0
- EnvySafari-Forge-1.1.5-1.16.5
- FastSuite-1.16.4-1.1.1
- FerriteCore-2.1.1-Forge
- FTBChunks-1605.3.4-build220
- FTBEssentials-1605-1.5-build.32
- FTBLibrary-1605.3.4-build.90
- FTBQuests-1605.1.6-build.33
- FTBTeams-1605.2.3-build.40
- FTBFilters-1605-2.5-build.0
- JEI-7.7.1.145
- JourneyMap-5.8.5p4
- Konkrete 1.6.1
- lazydfu-0.1.3
- lightspeed-1.0.4
- memoryleakfix-forge-pre1.17-1.0.0
- notenoughcrashes-4.1.4
- overworld_mirror-1.5.0.19
- pixelmon-9.1.5
- placebo-4.7.1
- proxy-compatible-forge-1.1.3 (for handling forge servers on Velocity, which we run)
- questsadditions 1.4.2
- rhino-forge-1605.1.5-build.75
- saturn-0.0.3
- spark-1.9.1
- starlight-1.2
- TrainerCommands 2.6.0
- u_team_core 3.2.1.229
- universalpokeloot-1.0.4
- voicechat-2.4.13
- waystones 7.6.4
- Wipeout-Faint-0.0.1
- worldedit-7.2.5

`[16:32:25] [Server thread/ERROR] [Minecraft/]: Could not pass event PlayerLoginEvent to EverNifeCore v2.0.3.3
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:303) ~[forge:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76) ~[forge:?]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[forge:?]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:593) ~[forge:?]
	at net.minecraft.server.management.PlayerList.canPlayerLogin(PlayerList.java:480) ~[?:?]
	at net.minecraft.network.login.ServerLoginNetHandler.func_147326_c(ServerLoginNetHandler.java:147) ~[?:?]
	at net.minecraft.network.login.ServerLoginNetHandler.func_73660_a(ServerLoginNetHandler.java:75) ~[?:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:239) ~[?:?]
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:142) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:1112) ~[?:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:339) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:1002) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:817) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:268) ~[?:?]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_312]
Caused by: java.lang.NullPointerException
	at br.com.finalcraft.evernifecore.listeners.PlayerLoginListener.onPlayerLogin(PlayerLoginListener.java:137) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_312]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_312]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_312]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_312]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:301) ~[forge:?]
	... 14 more`

**Comment by EverNife** [comment]
That is weird.

Use the command **/evernifecore info** and send me a print screen.

**Comment by andrewleppala** [comment]
![image](https://github.com/EverNife/EverNifeCore/assets/14192695/66eefbb3-8223-493d-a476-35a599d1a07e)

`evernifecore info
[06:09:23 INFO]: -----------------------------------------------------
[06:09:23 INFO]: # 1: ◆  BukkitPixelmonPlaceholders (1.1.1)
[06:09:23 INFO]: # 2: ◆  EverNifeCore (2.0.3.3)
[06:09:23 INFO]: 
[06:09:23 INFO]:                  <<<          Page [1/1]          >>>`

**Comment by EverNife** [comment]
The error is ocurring at the login of the player.

**Somehow a player is joining the server, and when the join process finish and my plugin is creating some cache, the PlayerData is not present.**

I can only think some kind of whitelist is present to cause this, can you tell me the context of the error? Does the error floods the console, does it happen rarely?

In your case this error will cause no problem, but if i can find the source of the problem its best to fix it.

**Comment by andrewleppala** [comment]
No gameplay error found, it's just flooding console (only once per login).

The server is connected via Velocity bridge, and they connect from a Lobby server via Mohist added Citizens2 NPC's calling CommandSync (https://www.spigotmc.org/resources/commandsync.115/) to ask Velocity to use Leaf (https://modrinth.com/plugin/leaf) to use /send to send the player to the appropriate server (this is because /server is created by Velocity and not properly registered, and consoles running that have no idea it's a command.)

The Lobby is whitelisted, Velocity is not (as theres no native whitelist and I never bothered installing a plugin), since Lobby is first in line in the joining process this keeps a solid barrier to unwanted players.

Low priority issue, since I don't believe it's actually causing any problems for me, but this may impact someone and wanted to report it.

**Comment by andrewleppala** [comment]
Update: Behavior does not appear to happen when running pure Forge, it seems to be brought about by either Mohist, or by some plugins I was loading with it.

**Comment by EverNife** [comment]
I am glad about that. Mohist is not the issue because i use it as well, so its probably some other plugin changing the default behavior of the login process. 

(That is why i asked about the whitelist thing, i have faced similar problems before with a bad whitelist system)

I will keep this issue open for a while. 
I will also try to adding an extra verification to prevent it from happening when another plugin do something bad.

**Comment by andrewleppala** [comment]
Most likely either CMI (Content Management Interface, a premium Spigot plugin that replaces the Essentials of old), or ProxyCompatibleForge mod (paired with Ambassador on the Velocity proxy) which adds some networking features that make Velocity work with modded clients/servers.

**Comment by EverNife** [comment]
Probably fixed at https://github.com/EverNife/EverNifeCore/commit/08a06bacbd740449034d2370677e549331edf466
